src/main/java/graphql/schema/GraphQLInputObjectField.java

```diff
@@ -170,14 +170,20 @@ public String toString() {
         return "GraphQLInputObjectField{" +
                 "name='" + name + '\'' +
                 ", description='" + description + '\'' +
-                ", originalType=" + originalType +
+                ", originalType=" + inputTypeToStringAvoidingCircularReference(originalType) +
                 ", defaultValue=" + defaultValue +
                 ", definition=" + definition +
                 ", directives=" + directives +
-                ", replacedType=" + replacedType +
+                ", replacedType=" + inputTypeToStringAvoidingCircularReference(replacedType) +
```

Comment:
This is a simple mechanism to break the infinite cycle. 
It could be implemented in either class. I've decided to the check in `GraphQLInputObjectField` for no particular reason.